### ISSUE TITLE
[Dedicated Server] Strange issues upon trying to cast a projectile spell

### ISSUE THREAD

**Issue by Febilian** [Growlith1223/ArsMagica2#15]
https://gist.github.com/Febilian/115b38880f1e2bb028b62d1960c20ed7

Nothing showed up when I cast the spell(projectile force damage, as basic as you can get), but my mana dropped down and then refused to regenerate itself naturally before I could try casting it again.

I thought at first it might be because I tried to cast it from my offhand, but the log seems to mention something about missing particles and a texture map, so now I'm not really sure what's going on.

So basically there are two things that were weird:

- The spell, when cast, had no particles or visuals show up, whereas in previous versions of AM2 for 1.7, you'd see the spell shoot out. I did not have a target, if it matters - I was casting at the ground a bit of a ways away in order to see whether it would work in my offhand.

- My mana did not visibly regenerate after this happened. It was stuck at 50 mana(in numerical terms) for a good minute or two, when I'd immediately picked the "mana regen I" talent in the occulus as one of my initial 3 blue skill points.

**Comment by Growlith1223** [comment]
this log is from the server, correct?

**Comment by Febilian** [comment]
Yes. I didn't see anything out of the ordinary on the client's log, so I checked the server's.

**Comment by Growlith1223** [comment]
hm, wonder why it would be trying to register icons on a server... lol i'll check it out, thanks for letting me know!

**Comment by Febilian** [comment]
Sure thing! Let me know if there's something more I can do to help as far as information goes. I'm really glad AM2 got ported over to 1.10.2, so thanks for taking the time to do this!

**Comment by Febilian** [comment]
So apparently logging out and then back into my server causes my mana to appear full again(500/500), so it might just not be visually updating/syncing, or it's never being used up serverside at all for all I know. After all, I tried casting the spell at a stationary chicken this time and nothing happened at all. I also tried without anything in my offhand slot, holding nothing but the spell - no dice. It seems that whatever is happening is preventing the spell from being properly cast at all.

**Comment by Growlith1223** [comment]
from what im seeing, AMParticleIcons isn't even being initialized..? lol ima have to look into this more

**Comment by Growlith1223** [comment]
scratch that it is

**Comment by Febilian** [comment]
Let me see if I can get a more detailed log of everything happening when the server is loading; maybe that will help.

**Comment by Growlith1223** [comment]
quick test but when try to force a high level, use up all your mana, then relog, and see what happens

**Comment by Febilian** [comment]
Can you rephrase that? I'm not sure what you mean, if you're asking me to test something.

**Comment by Growlith1223** [comment]
i don't remember the command but there's a command to set your ars magica level, i want you to set it to something high(like 100), cast a spell until your mana runs out, then log out, and log back in, and check and see if your mana is back to full.

**Comment by Febilian** [comment]
I can't get the command to work. 

Using this for reference: http://am2.wikia.com/wiki/Commands

/setmagiclevel febilian 100 does nothing except give me the message clientside that it did not find a valid command.

**Comment by Growlith1223** [comment]
ok it's /am2 magiclevel <number>

**Comment by Febilian** [comment]
Well. I cheated in a beam spell - fire damage with three solar modifiers, because I figured that'd drain my mana a lot faster than my dinky force damage projectile would - and I ended up getting a weird error serverside and disconnecting after a few seconds.

https://gist.github.com/Febilian/155493e6d6e3c83b8789468d1a8e0177

**Comment by Growlith1223** [comment]
gonna post a download that will hopefully fix that weird error

**Comment by Growlith1223** [comment]
ok, regarding the desync, i can reproduce that, haven't tested the other issues but i'll check them out as well

**Comment by Growlith1223** [comment]
so, the desync and the error are one in the same, still working on fixing it but i know what's causing it!

**Comment by Growlith1223** [comment]
both the disconnect with the burnout and the desync/error with particles has been fixed, will post new download soon(you can find the download under the releases section!)
